### PR TITLE
[pkg/stanza] use octetmultiline in input/syslog

### PR DESCRIPTION
**Pull request by timannguyen**
**Description:** <Describe what has changed.>

moved https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/operator/input/syslog/syslog.go#L64-L66 to be after https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/pkg/stanza/operator/input/syslog/syslog.go#L68 to ensure `OctetMultiLineBuilder` is being used when `enable_octet_counting` = `true`

**Link to tracking Issue:** [24073](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/24073)

**Testing:** 
manual test

**Documentation:** <Describe the documentation added.>
N/A

### CODE REVIEW

**Comment by timannguyen** [comment]
cc: @splunkericl

**Review by splunkericl — COMMENTED** [review]
can you add unit testing for that scenario?

**Review by splunkericl — APPROVED** [review]

File: pkg/stanza/operator/input/syslog/syslog_test.go [hunk 1]
```diff
@@ -21,14 +22,55 @@ import (
...
+		},
+		ValidForTCP: true,
+	}
+)
+
+func TestInput(t *testing.T) {
 
 	cases, err := syslog.CreateCases(basicConfig)
 	require.NoError(t, err)
+	cases = append(cases, OctetCase)
```
Comment (djaglowski) [inline pkg/stanza/operator/input/syslog/syslog_test.go:73]: Why not add the case to `CreateCases`?

Comment (djaglowski) [reply pkg/stanza/operator/input/syslog/syslog_test.go:73]: Ah, right. Thanks for reminding me.

**Review by djaglowski — APPROVED** [review]

File: .chloggen/pkg-stanza-input-syslog.yaml [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: bug_fix
+
+# The name of the component, or a single word describing the area of concern, (e.g. filelogreceiver)
+component: pkg/stanza/input/syslog
```
Comment (djaglowski) [inline .chloggen/pkg-stanza-input-syslog.yaml:9]:
```suggestion
component: pkg/stanza
```

File: .chloggen/pkg-stanza-input-syslog.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# A brief description of the change.  Surround your text with quotes ("") if it needs to start with a backtick (`).
+note: ensure octetmultiplinebuilder is initialized
```
Comment (djaglowski) [inline .chloggen/pkg-stanza-input-syslog.yaml:12]:
```suggestion
note: Fix issue where syslog input ignored enable_octet_counting setting
```

**Review by splunkericl — APPROVED** [review]

**Pull request merged**